### PR TITLE
Add debug logs to capture params used for memory calculation

### PR DESCRIPTION
**Pull request by yankevn**
This change adds debug logs for params used when calculating the allocated memory for merge tasks. These logs will help debugging efforts for nodes whose process memory consistently surpasses the allocated memory, thus implying the allocated memory calculation was incorrect.

### CODE REVIEW

**Review by raghumdani — COMMENTED** [review]
Two points:
1. Do you also want to log hash bucket memory calculation params?
2. In merge logs, we may need to add merge index so that we can easily connect.

**Review by raghumdani — COMMENTED** [review]
LGTM! Few minor comments.

File: deltacat/compute/compactor_v2/utils/task_options.py [hunk 1]
```diff
@@ -165,9 +171,17 @@ def hash_bucket_resource_options_provider(
...
+    debug_memory_params["size_bytes"] = size_bytes
+    debug_memory_params["num_rows"] = num_rows
+    debug_memory_params["total_pk_size"] = total_pk_size
+    debug_memory_params["total_memory"] = total_memory
 
     # Consider buffer
     total_memory = total_memory * (1 + TOTAL_MEMORY_BUFFER_PERCENTAGE / 100.0)
+    debug_memory_params["total_memory_with_buffer"] = total_memory
+    logger.debug(
+        f"Params used for calculating hash bucketing memory: {debug_memory_params}"
```
[the remark this answers is not in the document]
Comment (yankevn) [reply deltacat/compute/compactor_v2/utils/task_options.py:183]: Sure, makes sense

File: deltacat/compute/compactor_v2/utils/task_options.py [hunk 2]
```diff
@@ -133,11 +137,15 @@ def hash_bucket_resource_options_provider(
     ray_custom_resources: Optional[Dict] = None,
     **kwargs,
 ) -> Dict:
+    debug_memory_params = {"hash_bucket_task_index": index}
```
Comment (raghumdani) [inline deltacat/compute/compactor_v2/utils/task_options.py:140]: This is now redundant!

File: deltacat/compute/compactor_v2/utils/task_options.py [hunk 3]
```diff
@@ -186,10 +202,13 @@ def merge_resource_options_provider(
     deltacat_storage_kwargs: Optional[Dict] = {},
     **kwargs,
 ) -> Dict:
+    debug_memory_params = {"merge_task_index": index}
```
Comment (raghumdani) [inline deltacat/compute/compactor_v2/utils/task_options.py:205]: Same here, as this is now redundant.

File: deltacat/compute/compactor_v2/utils/task_options.py [hunk 4]
```diff
@@ -256,7 +278,14 @@ def merge_resource_options_provider(
...
     )
+    debug_memory_params["data_size"] = data_size
+    debug_memory_params["num_rows"] = num_rows
+    debug_memory_params["pk_size_bytes"] = pk_size_bytes
+    debug_memory_params["incremental_index_array_size"] = incremental_index_array_size
+    debug_memory_params["total_memory"] = total_memory
 
     total_memory = total_memory * (1 + TOTAL_MEMORY_BUFFER_PERCENTAGE / 100.0)
+    debug_memory_params["total_memory_with_buffer"] = total_memory
+    logger.debug(f"Params used for calculating merge memory: {debug_memory_params}")
```
Comment (raghumdani) [inline deltacat/compute/compactor_v2/utils/task_options.py:289]: Merge index not added here.

Comment (yankevn) [reply deltacat/compute/compactor_v2/utils/task_options.py:289]: Whoops, forgot to hit save 🙃

File: deltacat/compute/compactor_v2/utils/task_options.py (hunk 2 above)
Comment (yankevn) [reply deltacat/compute/compactor_v2/utils/task_options.py:140]: I think it's ok to keep these, it makes the printed objects self-identifiable.

Comment (raghumdani) [reply deltacat/compute/compactor_v2/utils/task_options.py:140]: I don't see them being read outside of the log. So, not sure why they have to self-identifiable. I have no strong bias either ways.

**Review by raghumdani — APPROVED** [review]
LGTM!

**Pull request merged**